.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -29,3 +29,16 @@ Regression? Last working version:
   - [ ] Documentation PR or issue filled
   - **OR**
   - [ ] N/A
+
+- **Pipeline**
```

Comment:
I feel like the completion rate of the current checklist is really low and something that many contributors ignore. 
I feel like this would just be seen as noise. 

I'd love to see our PR have a faster turnaround and I'm not sure a longer checklist does that.

Reply:
We have had this checklist for years, and I don't know why we have it.  Are you saying that because I want to call out something else in the checklist, it's time to reconsider it? What do you propose as an alternative?

Reply:
Here's some background: 

* The checklist has a purpose. It is heavily used by the release and release notes tools. https://github.com/NuGet/Entropy/blob/main/NuGetReleaseTool/NuGetReleaseTool/GenerateReleaseNotesCommand/README.md, https://github.com/NuGet/Entropy/blob/main/NuGetReleaseTool/NuGetReleaseTool/ValidateReleaseCommand/README.md and https://github.com/NuGet/Entropy/blob/main/NuGetReleaseTool/NuGetReleaseTool/GenerateReleaseChangelogCommand/README.md.

Let's look at it by sections:

Issue tidiness, 3 checkboxes
* It enforces a correlation between changes and issues, thus ensuring all issues are correctly included in the release notes.

Documentation:
* It also servers as a reminder for linking documentation, and has helped us get features documented before they ship: https://github.com/NuGet/Client.Engineering/blob/main/docs/Release-Manager.md#nugetclient-release-manager.

Tests:
* This is not used by automation right now, but it serves as a reminder of the test stragies. 

As minimal as it is, the 3 sections tend to apply to all fixes in large majority of the time.

As a code reviewer, I frequently have to ask people to complete the checklist. 
Heck, some people flat out delete it, despite the first line saying `<!-- DO NOT MODIFY OR DELETE THIS TEMPLATE. IT IS USED IN AUTOMATION. -->`

Adding checkboxes for something that'd be used infrequently (most changes are product changes), would make the checklist even more intimidating. 

I think this reminder is valuable, but maybe something better served in a readme of the eng folder or something like that. 

Re: changes to the current template, I think that's a slightly different discussion, and I do have some ideas I can propose in isolation, but they wouldn't be a big change, rather one that reduces the details a contributor needs to fill out.